dagfactory/dagbuilder.py

```diff
@@ -324,6 +351,45 @@ def _clean_kpo_task_params(task_params: dict) -> dict:
 
         return task_params
 
+    @staticmethod
+    def _handle_http_sensor(operator_obj, task_params):
+        # Only handle if HttpOperator/HttpSensor are available
+        if HTTP_OPERATOR_CLASS and issubclass(operator_obj, HTTP_OPERATOR_CLASS):
+            headers = task_params.get("headers", {})
+            content_type = headers.get("Content-Type", "").lower()
+
+            if "data" in task_params and "application/json" in content_type:
+                task_params["data"]: Callable = utils.get_json_serialized_callable(task_params["data"])
+
+                if "Content-Type" not in headers:
+                    headers["Content-Type"] = "application/json"
+                task_params["headers"] = headers
+        elif HTTP_SENSOR_CLASS and issubclass(operator_obj, HTTP_SENSOR_CLASS):
+            if not (
+                task_params.get("response_check_name") and task_params.get("response_check_file")
+            ) and not task_params.get("response_check_lambda"):
+                raise DagFactoryException(
+                    "Failed to create task. HttpSensor requires \
+                    `response_check_name` and `response_check_file` parameters \
+                    or `response_check_lambda` parameter."
+                )
+            if task_params.get("response_check_file"):
+                task_params["response_check"]: Callable = utils.get_python_callable(
+                    task_params["response_check_name"], task_params["response_check_file"]
+                )
+                # remove dag-factory specific parameters
+                # Airflow 2.0 doesn't allow these to be passed to operator
+                del task_params["response_check_name"]
+                del task_params["response_check_file"]
```

Comment:
We may be able to simplify the code by doing something similar to this:

```suggestion
            # remove dag-factory specific parameters
            # Airflow 2.0 doesn't allow these to 
            response_check_name = task_params.pop("response_check_name", None)
            response_check_file = task_params.pop("response_check_file", None)
            if response_check_name
                task_params["response_check"]: Callable = utils.get_python_callable(
                    response_check_name, response_check_file
                )
```

This suggestion will not work out of the box because we're in the middle of an if-else statement, but the relevant part is to leverage `pop`.